Fixed PGVector issue when SQLAlchemy is the driver

ranaroussi opened the pull request:
Using SQLAlchemy as the driver for PGVector's `connection_string_from_db_params` function fails with the following error:

```
NoSuchModuleError: Can't load plugin: sqlalchemy.dialects:postgresql.sqlalchemy
```

This PR removes the `+{driver}` when `sqlalchemy` is used as the driver.

Additionally, this PR also adds support for query params in the connection string via the optional `query_params` param. 

Example:

```python
CONNECTION_STRING = PGVector.connection_string_from_db_params(
    driver="sqlalchemy", 
    host="dbhost.com",
    port=5432,
    database="dbname",
    user="user",
    password="pass",
    query_params={ # ← this is new
        "sslmode": "require"
    }
)'

print(CONNECTION_STRING)

# 'postgresql://user:pass@dbhost.com:5432/dbname?sslmode=require', 
```

Tested on Postgres 14, 15, and Supabase

@dev2049 ☝️

ranaroussi committed:
Updated connection_string_from_db_params to support query params and sqlalchemy

Using `postgresql+sqlalchemy` as driver fails to connect. This commit removes the `+{driver}` when `sqlalchemy` is used the driver.

Additionally, support for query params as `kwargs` are now added to the connection string (eg. `?sslmode=require`

baskaryan commented on langchain/vectorstores/pgvector.py:
```diff
@@ -593,6 +594,9 @@ def connection_string_from_db_params(
         database: str,
         user: str,
         password: str,
+        **kwargs
```
should we just call this query_params and make it an optional dict if that's what it's used for

ranaroussi replied:
That's a great idea. I'll push an updated PR

ranaroussi committed:
Replaced kwargs with query_params (optional dict)

Following suggestion from @baskaryan https://github.com/hwchase17/langchain/pull/6377/files#r1254724983

leo-gan commented:
@ranaroussi Hi , could you, please, resolve the merging issues? After that ping me and I push this PR for the review. Thanks!

efriis commented:
Closing because the PR wouldn't line up with the current directory structure of the library (would need to be in /libs/langchain/langchain instead of /langchain). Feel free to reopen against the current head if it's still relevant!
